File: javascript/decorator/StarbuzzCoffee/decorator/condiments/Soy.js
import {CondimentDecorator} from "../CondimentDecorator.js";
import {Beverage} from "../../component/Beverage.js";

export class Soy extends CondimentDecorator {

    beverage = new Beverage()

    constructor(beverage) {
        super();
        this.beverage = beverage
    }

    getDescription() {
        return this.beverage.getDescription() + ", Soy"
    }

    cost() {
        return this.beverage.cost() + 0.22
    }

}

File: C++/Decorator/StarbuzzCoffee/decorator/condiments/Soy.cpp
#include "../CondimentDecorator.h"

class Soy : public CondimentDecorator {

public:
    explicit Soy(Beverage* beverage) { this->beverage = beverage; }

    string getDescription() override { return beverage->getDescription() + ", Soy"; }

    double cost() override { return beverage->cost() + 0.20; }

private:
    Beverage* beverage;

};

